sct.py

```diff
@@ -484,6 +487,79 @@ def list_repos(dist_type, dist_version):
+              help='Scylla repo')
+@click.option('-d', '--linux-distro', type=str,
+              default='centos', help='Linux Distribution type')
+@click.option('-o', '--only-print-versions', type=bool, default=False, required=False, help='')
+def get_scylla_base_versions(scylla_version, scylla_repo, linux_distro, only_print_versions):  # pylint: disable=too-many-locals
+    """
+    Upgrade test try to upgrade from multiple supported base versions, this command is used to
+    get the base versions according to the scylla repo and distro type, then we don't need to hardcode
+    the base version for each branch.
+    """
+    add_file_logger()
+
+    dist_type = linux_distro.split('-')[0]
+    dist_version = linux_distro.split('-')[-1]
+
+    if scylla_version is None:
+        assert 'unstable/' in scylla_repo
+        product, version = scylla_repo.split('unstable/')[1].split('/')[0:2]
+        scylla_version = version.replace('branch-', '').replace('enterprise-', '')
+    else:
+        product = 'scylla-enterprise' if is_enterprise(scylla_version) else 'scylla'
+
+    if not dist_type == 'centos' and dist_version is None:
+        click.secho("when passing --dist-type=debian/ubuntu need to pass --dist-version as well", fg='red')
+        sys.exit(1)
+
+    # We can't detect the support versions for this distro, which shares the repo with others, eg: centos8
+    # so we need to assign the start support versions for it.
+    oos_start_support_version = None
+    ent_start_support_version = None
+    if (dist_type not in ['debian', 'ubuntu'] and linux_distro != 'centos') and (
+            oos_start_support_version is None or ent_start_support_version is None):
+        versions_dict = start_support_versions.get(linux_distro, None)
+        if versions_dict is None:
+            click.secho(
+                f"We can't detect the support versions for this distro, which shares the repo with {dist_type}.")
+            sys.exit(1)
+        oos_start_support_version = versions_dict['scylla']
+        ent_start_support_version = versions_dict['enterprise']
+
+    repo_maps = get_s3_scylla_repos_mapping(dist_type, dist_version)
+
+    supported_versions = []
+    # Filter out the unsupported versions
+    for version_prefix, _ in repo_maps.items():
+        if not re.match(r'\d+.\d+', version_prefix):
+            continue
+        if oos_start_support_version and not is_enterprise(version_prefix) and \
+                parse_version(version_prefix) < parse_version(oos_start_support_version):
+            continue
+        if ent_start_support_version and is_enterprise(version_prefix) and \
+                parse_version(version_prefix) < parse_version(ent_start_support_version):
+            continue
+        supported_versions.append(version_prefix)
+    click.echo(f'Supported Versions: {supported_versions}')
+    version_list = get_supported_scylla_base_versions(product, scylla_version, supported_versions)
```

Comment:
Please place this code it into the function and put it into `utils/get_supported_scylla_base_versions.py`